WebApp/src/uk/ac/exeter/QuinCe/web/files/UploadedDataFile.java

```diff
@@ -42,29 +41,11 @@ public UploadedFile getUploadedFile() {
   public void setUploadedFile(UploadedFile uploadedFile) {
     this.uploadedFile = uploadedFile;
     String[] fileLines = getLines();
-    rowCount = fileLines.length;
-    while ("".equals(fileLines[rowCount - 1])) {
```

Comment:
There's a couple of other possible things to check:

* Lines that contain only whitespace
* Lines that contain only the separator character (i.e. every column value is blank)

I think we can treat both of these as empty lines. Whether the check should be here or in `getLines` I'm not sure.

Reply:
Not sure if I understand: I'm removing all refs to rowCount in this commit, and rather using datafiles own recordCount - field. This should be correct. If not, that is another issue. Your comments seem to apply to the old rowCount - approach?

Reply:
Oh, sorry. I got confused between this and the other PR. I shouldn't be doing this after filling my face with ibuprofen!